cmake/BLTPrivateMacros.cmake

```diff
@@ -225,27 +280,95 @@ macro(blt_setup_openmp_target)
         message ( FATAL_ERROR "Must provide a BUILD_TARGET argument to the macro")
     endif()
 
-    if ( NOT DEFINED arg_USE_OPENMP )
-        message( FATAL_ERROR "Must provide an OpenMP boolean flag")
-    endif()
 
-    if ( ${arg_USE_OPENMP} AND NOT ${ENABLE_OPENMP} )
-        message( FATAL_ERROR "Building an OpenMP library, but OpenMP is disabled!")
+    if ( NOT ${ENABLE_OPENMP} )
+        message( FATAL_ERROR "Building target '${arg_BUILD_TARGET}' requires OpenMP, but OpenMP is disabled!")
     endif()
 
-    if ( ${arg_USE_OPENMP} )
-        blt_add_target_definitions( TO ${arg_BUILD_TARGET}
+    blt_add_target_definitions( TO ${arg_BUILD_TARGET}
                                 TARGET_DEFINITIONS USE_OPENMP )
-        set_target_properties( ${arg_BUILD_TARGET}
-                               PROPERTIES COMPILE_FLAGS ${OpenMP_CXX_FLAGS} )
-        set_target_properties( ${arg_BUILD_TARGET}
-                               PROPERTIES LINK_FLAGS ${OpenMP_CXX_FLAGS} )
-    endif()
+
+    blt_add_target_compile_flags( TO ${arg_BUILD_TARGET}
```

Comment:
Do we know if this works when using CUDA? I'm not sure if target_compile_flags works with cuda executables.